marshal.go

```diff
@@ -80,6 +80,9 @@ func (u *UUID) UnmarshalText(data []byte) error {
 // UnmarshalJSON reads an UUID from a JSON-string into the UUID instance.
 // If this fails the state of the UUID is undetermined.
 func (u *UUID) UnmarshalJSON(data []byte) error {
+	if len(data) == 0 {
+		return nil
```

Comment:
An empty byte slice or 1-2 bytes long slice is technically an invalid UUID and should result in an error. I have a proper fix on the way which will error if the byte slice is too short or not surrounded by quotes.